### PR TITLE
auth: ensure the redirect URL always starts with a single slash

### PR DESCRIPTION
**Pull request by unknwon**
Fixes https://github.com/sourcegraph/security-issues/issues/63

### CODE REVIEW

File: cmd/frontend/auth/redirect.go [hunk 1]
```diff
@@ -15,6 +15,9 @@ func SafeRedirectURL(urlStr string) string {
 		return "/"
 	}
 
+	// Make sure u.Path always starts with a single slash.
+	u.Path = "/" + strings.TrimLeft(u.Path, "/")
```
Comment (keegancsmith) [inline cmd/frontend/auth/redirect.go:19]: Alternatively this does a bit more:
```suggestion
	u.Path = path.Clean(u.Path)
```

Comment (unknwon) [reply cmd/frontend/auth/redirect.go:19]: Nice, thanks!

**Comment by nicksnyder** [comment]
I think this deserves an entry in our changelog, what do you think?

**Comment by nicksnyder** [comment]
Oh I see that you updated the changelog in a separate PR.

**Pull request merged**